habito/habito.py

```diff
@@ -103,9 +115,18 @@ def list():
 
             if quanta >= habit.quantum:
                 column_text = u'\u2713'
-
-            habit_row.append("{0} ({1})".format(column_text, quanta))
-
+                if in_streak:
```

Comment:
As for the complexity issue you raised, I definitely agree. I think we should apply the 'complex pattern', as shown here, https://github.com/pallets/click/tree/master/examples/complex, and also have a models and utilities file. From there on, we can start building these new features in a more sensible manner and start separating the view and the model, as you suggested. If you agree, I'm definitely willing to work on this next week.

I think it's best for streak to be calculated each time, since it can change so often. It's definitely possible to do this more efficiently I do like the `AchievementsModel` suggestion though, but it would be more useful for badges etc. (e.g. 5 checkins, 10 checkins, 5-day streak (at least once), ...).